examples/Recipies/initialization/modules/my_dataclass.py

```diff
@@ -0,0 +1,23 @@
+"""Module with two configurable class variables and two configurable module globals
+
+Configuration has been done such that the_first_classvar and the_first_global will be
+initialized with the default value (0.0) and not with the configured value (1.0);
+the_second_classvar and the_second_global will be initialized correctly with the
+configured value (1.0).
+"""
+
+from dataclasses import dataclass
+
+from modules.correctly_initialized_config import CorrectlyInitializedConfig
+from modules.wrongly_initialized_config import WronglyInitializedConfig
+
+
+@dataclass
+class MyDataclass:
+    the_first_classvar: float = WronglyInitializedConfig.get().initial_value
+    the_second_classvar: float = CorrectlyInitializedConfig.get().initial_value
+
+
+print("Now initializing the module globals")
+the_first_global = WronglyInitializedConfig.get().initial_value
```

Comment:
In the end, is the big problem not that we're doing a `get` before we have called a `load`? I think we should actually easily be able to detect that, and signal a warn, maybe even throw an error that we don't want that?

We could add some `implicit_load_allowed` parameter somewhere maybe to avoid the error if the user explicitly allows for the defaults to just be used.

Reply:
I refactored all file operations, they are now in a separate, private module. Warning if we're doing `get` before `load` has improved.

I think the `implicit_load_allowed` parameter is maybe not a good idea. It could work as an attribute that you define in the container, but if you want to change it with a method you are already too late, because such a method call will always be after importing. Let's not add that complexity.